app/services/streaming/streaming.ts

```diff
@@ -982,6 +1061,308 @@ export class StreamingService
+            this.createRecording('vertical', 2);
+          } else {
+            // Otherwise, record both displays in dual output mode
+            this.createRecording('vertical', 2);
+            this.createRecording('horizontal', 1);
+          }
+        } else {
+          // In single output mode, recording only the horizontal display
+          // TODO: remove after migrate output and stream settings to new api
+          this.recordWithOldAPI();
+          // this.createRecording('horizontal', 1);
+        }
+      }
+    }
+  }
+
+  private createRecording(display: TDisplayType, index: number) {
+    const mode = this.outputSettingsService.getSettings().mode;
+
+    const recording =
+      mode === 'Advanced'
+        ? (AdvancedRecordingFactory.create() as IAdvancedRecording)
+        : (SimpleRecordingFactory.create() as ISimpleRecording);
+
+    const settings =
+      mode === 'Advanced'
+        ? this.outputSettingsService.getAdvancedRecordingSettings()
+        : this.outputSettingsService.getSimpleRecordingSettings();
+
+    // assign settings
+    Object.keys(settings).forEach(key => {
+      if (key === 'encoder') {
+        recording.videoEncoder = VideoEncoderFactory.create(settings.encoder, 'video-encoder');
+      } else {
+        recording[key] = settings[key];
+      }
+    });
+
+    // assign context
+    recording.video = this.videoSettingsService.contexts[display];
+
+    // set signal handler
+    recording.signalHandler = async signal => {
+      await this.handleRecordingSignal(signal, display);
+    };
+
+    // handle unique properties (including audio)
+    if (mode === 'Advanced') {
+      recording['outputWidth'] = this.dualOutputService.views.videoSettings[display].outputWidth;
+      recording['outputHeight'] = this.dualOutputService.views.videoSettings[display].outputHeight;
+
+      const trackName = `track${index}`;
+      const track = AudioTrackFactory.create(160, trackName);
+      AudioTrackFactory.setAtIndex(track, index);
+    } else {
+      recording['audioEncoder'] = AudioEncoderFactory.create();
+    }
+
+    // save in state
+    if (display === 'vertical') {
+      this.verticalRecording = recording;
+      this.verticalRecording.start();
+    } else {
+      this.horizontalRecording = recording;
+      this.horizontalRecording.start();
+    }
+  }
+
+  private async handleRecordingSignal(info: EOutputSignal, display: TDisplayType) {
+    // map signals to status
+    const nextState: ERecordingState = ({
+      [EOutputSignalState.Start]: ERecordingState.Recording,
+      [EOutputSignalState.Stop]: ERecordingState.Offline,
+      [EOutputSignalState.Stopping]: ERecordingState.Stopping,
+      [EOutputSignalState.Wrote]: ERecordingState.Wrote,
+    } as Dictionary<ERecordingState>)[info.signal];
+
+    // We received a signal we didn't recognize
+    if (!nextState) return;
```

Comment:
I think logging out (or throwing errors on) these `will never happen` events would be good as we'd have the peace of mind it'll never silently happen.